### PR TITLE
Less restrictive visualizability constraints of 2d entitites, improved space view generation heuristics

### LINKED ISSUE #4926: Re-visit "child-of-root" space view spawning heuristics
Largely a follow-up of:

* #4388

There's a bunch of cases where we treat child entities of the root special for the purpose of spawning space views:

* 3D space views
   * NOTE: #5087 removes this for 2D views 
* time series space views
   * Spawns space views iff there's several child of root which don't have timeseries indicators themselves but have children that do
   * See `plots` demo why this is needed: We expect separate plots for each child of the root node. Each plot in turn contains several entities

This is problematic since it's a fairly complicated heuristic both for a user to understand and for our implementation to maintain!
On the flip side it allows to easily crate many space views without user ui interaction & logging anything extra!

Mitigation / strategies for removal:
* Spawn a 3D space view wherever there's `ViewCoordinate`, don't spawn in the "remaining space" if this wouldn't add any more entities
* Plots seems harder: We could just accept a single plot or a plot for each entity (this breaks the plots demo as-is)
* ... TODO: would the above be enough?

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #4926
   * it does not address child-of-root heuristics for time series, but they weren't as much of a concern
* Fixes heuristic issues observed in https://github.com/rerun-io/cpp-example-opencv-eigen/pull/20

Two changes in here:
* we no longer determine the dimensionality of a topological subspace, instead we only determine whether it can't display 2d/3d
   * 2d: we now state we can _always_ display 2d objects
   * 3d: we can display it unless the parent of the active subspace is a pinhole in which case we only make the entity _at_ the origin as visualizable
* make `ViewCoordinates` add "heuristic hints" to spatial topology
   * this is all that's needed to get rid of child-of-root heuristics in 3d without regressing existing examples
   
OpenCV example that didn't work before:
image

New release check:
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5188/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5188/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5188/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5188)
- [Docs preview](https://rerun.io/preview/cc3735c02cd092df5843f7fe97bfa998db2e459d/docs) 
- [Examples preview](https://rerun.io/preview/cc3735c02cd092df5843f7fe97bfa998db2e459d/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by Wumpf** [comment]
converted to draft: just now tested _all_ `nightly` examples on the web build and turns out that structure from motion creates an extra unwanted space view at `camera`

**Comment by Wumpf** [comment]
fixed, that was actually a pretty evil bug. added a test for it

File: crates/re_space_view_spatial/src/space_view_3d.rs [hunk 1]
```diff
@@ -84,46 +83,40 @@ impl SpaceViewClass for SpatialSpaceView3D {
...
-                SubSpaceDimensionality::TwoD => {
-                    // If this is 2D space, only display the origin entity itself.
-                    // Everything else we have to assume requires some form of transformation.
-                    VisualizableFilterContext3D {
-                        entities_in_main_3d_space: std::iter::once(space_origin.clone()).collect(),
-                        entities_under_pinholes: Default::default(),
-                    }
-                }
+            VisualizableFilterContext3D {
+                entities_in_main_3d_space,
```
Comment (Wumpf) [inline crates/re_space_view_spatial/src/space_view_3d.rs:118]: this should recursively collect all entities that are in child spaces that are not excluding 3D spaces.
this recursion needs to go both down the children and up the parent.

Similarly we also need to collect more pinholes!

**Review by Wumpf — COMMENTED** [review]
from discussion on the phone:
* don't split via ViewCoordinates, this makes visualizability context a lot more complex
* instead note "interesting" roots within a SubSpace down (in SpatialTopology) so we can use this in the spawn heuristics
    * future work: we could so similar things for 2D?

**Review by jleibs — APPROVED** [review]

**Pull request merged**